File: Java/ValidAnagram.java
package Java;

import java.util.Arrays;

public class ValidAnagram {
  private static boolean isAnagram(String a, String b) {
    if (a.length() != b.length()) return false;

    char[] arrayA = a.toCharArray();
    char[] arrayB = b.toCharArray();

    Arrays.sort(arrayA);
    Arrays.sort(arrayB);

    return Arrays.equals(arrayA, arrayB);
  }

  public static void main(String[] args) {
    System.out.println(isAnagram("anagram", "nagaram"));
    System.out.println(isAnagram("rat", "car"));
  }
}

File: TS/validAnagram.ts
function isAnagram(firstWord: string, secondWord: string): boolean {
  if (firstWord.length !== secondWord.length) return false;

  const firstArray = firstWord.toLowerCase().split("");
  const secondArray = secondWord.toLowerCase().split("");

  return firstArray.sort().join("") === secondArray.sort().join("");
}

console.log(isAnagram("anagram", "nagaram"));
console.log(isAnagram("cat", "rat"));

